rockcraft/extensions/flask.py

```diff
@@ -0,0 +1,169 @@
+        snippet["services"] = self._gen_services()
+        return snippet
+
+    def _gen_services(self):
+        """Return the services snipped to be applied to the rockcraft file."""
+        services = {
+            "flask": {
+                "override": "replace",
+                "startup": "enabled",
+                "command": "/bin/python3 -m gunicorn --bind 0.0.0.0:8000 app:app",
+                "user": "_daemon_",
+                "working-dir": "/srv/flask/app",
+            }
+        }
+        existing_services = self.yaml_data.get("services", {})
+        for existing_service_name, existing_service in existing_services.items():
+            if existing_service_name in services:
+                services[existing_service_name].update(existing_service)
+            else:
+                services[existing_service_name] = existing_service
+        return services
+
+    @override
+    def get_part_snippet(self) -> Dict[str, Any]:
+        """Return the part snippet to apply to existing parts."""
+        return {}
+
+    def _merge_part(self, base_part: dict, new_part: dict) -> dict:
+        """Merge two part definitions by the extension part merging rule."""
+        result = {}
+        properties = set(base_part.keys()).union(set(new_part.keys()))
+        for property_name in properties:
+            if property_name in base_part and property_name not in new_part:
+                result[property_name] = base_part[property_name]
+            elif property_name not in base_part and property_name in new_part:
+                result[property_name] = new_part[property_name]
+            else:
+                result[property_name] = _apply_extension_property(
+                    base_part[property_name], new_part[property_name]
+                )
+        return result
+
+    def _merge_existing_part(self, part_name: str, part_def: dict) -> dict:
+        """Merge the new part with the existing part in the current rockcraft.yaml."""
+        existing_part = self.yaml_data.get("parts", {}).get(part_name, {})
+        return self._merge_part(existing_part, part_def)
+
+    def _gen_new_parts(self) -> Dict[str, Any]:
+        """Generate new parts for the flask extension.
+
+        Parts added:
+            - flask/dependencies: install Python dependencies
+            - flask/install-app: copy the flask project into the OCI image
+        """
+        if not (self.project_root / "requirements.txt").exists():
+            raise ValueError(
+                "missing requirements.txt file, "
+                "flask extension requires this file with flask specified as a dependency"
+            )
+        ignores = [".git", "node_modules", ".yarn"]
+        source_files = [
+            f
+            for f in os.listdir(self.project_root)
+            if f not in ignores and not f.endswith(".rock")
+        ]
+        renaming_map = {f: posixpath.join("srv/flask/app", f) for f in source_files}
```

Comment:
I was trying to find the handling of requiring prime and checking that everything in there has the `srv/flask/app` prefix?

Reply:
That has been added in the latest commit, thanks!